### ISSUE TITLE
Cells with custom date format not converted correctly or throw error

### ISSUE THREAD

**Issue by gorj-tessella** [Apkawa/xlsx2html#26]
When attempting to process a sheet with a date in the custom format "d-mmm-yyyy", it failed with a "ValueError: Invalid length for field 'mmm'". The error was thrown by babel. 

Upon further inspection I found that pyxl was using the cell's number_format property which was "d-mmm-yyyy". When it ran the date conversions function, nothing was changed. Babel needed this to be converted to "d-MMM-yyyy". I found that the normalize_date_format was not doing this; it was making the following changes: 

https://github.com/Apkawa/xlsx2html/blob/bbd384341be2920c8782dbe60467d531b851da86/xlsx2html/format/dt.py#L8-L12

The root issues are as follows: 

- dt.py expects dates to be written with upper case for some reason. Maybe that is true for internal formats, but it isn't for custom formats. The "mmm" needed to be converted to "MMM"
- Custom date/time formats in excel are frustratingly ambiguous. Specifically, both month and minute are specified with lowercase "m". "m" is only interpreted as minute if follows "hh" or precedes "ss". 
- Openpyxl doesn't provide any capabilities to get the value as formatted from a cell, though it really should. 

Reference: 
- [Babel date formats](http://babel.pocoo.org/en/latest/dates.html)
- [Excel date formats](https://support.microsoft.com/en-us/office/format-numbers-as-dates-or-times-418bd3fe-0577-47c8-8caa-b4d30c528309)

**Comment by gorj-tessella** [comment]
The following code should be sufficient to format an openpyxl datetime object using an excel format, based on excel's definitions. 

```
days = [
    'Monday', 'Tuesday', 'Wednesday',
    'Thursday', 'Friday', 'Saturday', 'Sunday'
]

months = [
    'January', 'February', 'March', 'April',
    'May', 'June', 'July', 'August',
    'September', 'October', 'November', 'December'
]

date_tok_rx = re.compile(r"yyyy|yy|m{1,5}|d{1,4}|h{1,2}|s{1,2}|\.0+|AM/PM|A/P")


def zpad(v, tok):
    v = str(v)
    i = len(tok) - len(v)
    if i > 0:
        v = ('0' * i) + v
    return v


def excel_format_date(fmt, value):
    has_ap = False
    maybe_minute = {}
    parts = []
    toks = []
    pos = 0
    for m in date_tok_rx.finditer(fmt):
        tok = m.group(0)
        start, end = m.span(0)
        parts.append(fmt[pos:start])
        i = len(parts)
        parts.append(None)
        if tok in ['AM/PM' or 'A/P']:
            has_ap = True
        elif tok[0] == 'h':
            maybe_minute.add(i + 1)
        elif tok[0] == 's':
            maybe_minute.add(i - 1)
        toks.append((tok, i))
        pos = end
    parts.append(fmt[pos:])

    for tok, i in toks:
        if tok in ['h', 'hh']:
            v = value.hour
            if has_ap:
                v = v % 12
                if v == 0:
                    v = 12
            parts[i] = zpad(v, tok)
        elif tok in ['m', 'mm'] and i in maybe_minute:
            parts[i] = zpad(value.minute, tok)
        elif tok in ['s', 'ss'] and i in maybe_minute:
            parts[i] = zpad(value.second, tok)
        elif tok[:2] == '.0':
            digits = len(tok) - 1
            v = value.millisecond / 1000000.0
            v = ("{." + str(digits) + "f}").format(v)[1:]
            parts[i] = v
        elif tok == 'AM/PM':
            parts[i] = 'AM' if (value.hour < 12) else 'PM'
        elif tok == 'A/P':
            parts[i] = 'A' if (value.hour < 12) else 'P'
        elif tok == 'yyyy':
            parts[i] = zpad(value.year, tok)
        elif tok == 'yy':
            parts[i] = zpad(value.year % 100, tok)
        elif tok in ['m', 'mm']:
            parts[i] = zpad(value.month, tok)
        elif tok == 'mmm':
            parts[i] = months[value.month - 1][:3]
        elif tok == 'mmmm':
            parts[i] = months[value.month - 1]
        elif tok == 'mmmmm':
            parts[i] = months[value.month - 1][0]
        elif tok in ['d', 'dd']:
            parts[i] = zpad(value.day, tok)
        elif tok == 'ddd':
            parts[i] = days[value.weekday()][:3]
        elif tok == 'dddd':
            parts[i] = days[value.weekday()]
        else:
            raise ValueError(f'Illegal date tok {tok}')

    print(parts)
    return ''.join(parts)
```

**Comment by Apkawa** [comment]
Great job! I will check soon and add your solution as soon as there is time.
The current implementation was incomplete and only covered my tasks.

**Comment by Apkawa** [comment]
published to v0.3.0
